dbms/src/TestUtils/mockExecutor.h

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Debug/astToExecutor.h>
+#include <Interpreters/Context.h>
+
+#include <cstddef>
+#include <initializer_list>
+#include <utility>
+
+namespace DB
+{
+namespace tests
+{
+using MockColumnInfo = std::pair<String, TiDB::TP>;
+using MockColumnInfos = std::vector<MockColumnInfo>;
+using MockColumnInfoList = std::initializer_list<MockColumnInfo>;
+using MockTableName = std::pair<String, String>;
+using MockOrderByItem = std::pair<String, bool>;
+using MockOrderByItems = std::initializer_list<MockOrderByItem>;
+using MockColumnNames = std::initializer_list<String>;
+using MockAsts = std::initializer_list<ASTPtr>;
+
+class DAGRequestBuilder
+{
+public:
+    static size_t executor_index;
```

Comment:
a bit hack..

maybe we can make a `builder` of `DAGRequestBuilder` to hold `executor_index`..